eslint-plugin/src/rules/4/deprecated.class-methods.ts

```diff
@@ -0,0 +1,30 @@
+import {lte} from 'semver';
+import {ESL_PACKAGE_VERSION} from '../../core/check-version';
+import {buildLoggingRule} from '../../core/log';
+import {buildRule} from '../../core/deprecated-class-method';
+
+import type * as ESTree from 'estree';
+import type {ESLintReplacementMethodCfg} from '../../core/deprecated-class-method';
+
+const AVAILABLE_SINCE = '5.0.0-beta.24';
+const isActual = lte(ESL_PACKAGE_VERSION, AVAILABLE_SINCE);
+
+/**
+ * Rule for deprecated methods
+ */
+export default buildRule({
+  className: 'ESLMediaRuleList',
+  deprecatedMethod: 'parse',
+  getReplacementMethod: (expression): ESLintReplacementMethodCfg | string => {
+    const args = expression.arguments;
+    const isLiteral = (node: ESTree.Expression | ESTree.SpreadElement): boolean => node?.type === 'Literal' || node?.type === 'TemplateLiteral';
+    if (expression.type === 'CallExpression' && args.length === 1) return 'parseQuery';
+    if (expression.type === 'CallExpression' && args.length === 2 && isLiteral(args[1])) return 'parseTuple';
+    if (expression.type === 'CallExpression' && args.length === 3) return 'parseTuple';
+    return {message: 'ESLMediaRuleList.parseQuery or ESLMediaRuleList.parseTuple'};
+  },
+  skipOn: () => {
+    !isActual && buildLoggingRule(`'ESLMediaRuleList.parse' was updated in v${AVAILABLE_SINCE}. Rule 'deprecated-4/media-rule-list-parse' is skipped.`);
```

Comment:
buildLogingRule will not be executed as it is builder that returns dummy rule, consider replacing with the log or abstract log in case skipOn rejected.
As alternative you can also consider ext signature for skipOn:
```
   skipOn() {
      // if (isActual) return false;
      if (!isActual) return {
         actual: false,
         message: `'ESLMediaRuleList.parse' was updated in v${AVAILABLE_SINCE}. Rule 'deprecated-4/media-rule-list-parse' is skipped.`
      };
   } 
```